src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsTransport.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\AllMySMs;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Quentin Dequippe <quentin@dequippe.tech>
+ *
+ * @experimental in 5.3
+ */
+final class AllMySmsTransport extends AbstractTransport
+{
+    protected const HOST = 'api.allmysms.com';
+
+    private $login;
+    private $apiKey;
+    private $from;
+
+    public function __construct(string $login, string $apiKey, string $from = null, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $login;
+        $this->apiKey = $apiKey;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('allmysms://%s?from=%s', $this->getEndpoint(), $this->from);
```

Comment:
From parameter should only be added if not null

Reply:
I have to remove `?from=%s`?